tests/integration/workflows/test_arxiv_workflow.py

```diff
@@ -498,3 +503,109 @@ def test_article_workflow_continues_when_record_is_valid(workflow_app):
+    obj_id = WorkflowEngine.from_uuid(eng_uuid).objects[0].id
+    obj = workflow_object_class.get(obj_id)
+
+    assert obj.extra_data['already-in-holding-pen'] is False
+    assert obj.extra_data['holdingpen_matches'] == []
+    assert obj.extra_data['previously_rejected'] is False
+    assert not obj.extra_data.get('stopped-matched-holdingpen-wf')
+    assert obj.extra_data['is-update']
+    assert obj.extra_data['exact-matched']
+    assert obj.extra_data['matches']['exact'] == [record.get('control_number')]
+    assert obj.extra_data['matches']['approved'] == record.get('control_number')
+
+
+@mock.patch(
+    'inspirehep.modules.workflows.tasks.magpie.json_api_request',
+    side_effect=fake_magpie_api_request,
+)
+@mock.patch(
+    'inspirehep.modules.workflows.tasks.beard.json_api_request',
+    side_effect=fake_beard_api_request,
+)
+@mock.patch(
+    'inspirehep.modules.workflows.tasks.arxiv.download_file_to_workflow',
+    side_effect=fake_download_file,
+)
+@mock.patch(
+    'inspirehep.modules.workflows.tasks.arxiv.is_pdf_link',
+    return_value=True
+)
+def test_fuzzy_matched_goes_trough_the_workflow(
+    mocked_is_pdf_link,
+    mocked_download_arxiv,
+    mocked_api_request_beard,
+    mocked_api_request_magpie,
+    workflow_app,
+    mocked_external_services,
+    record_from_db,
+):
+    """Test update article fuzzy matched.
+
+    In this test the `WorkflowObject.continue_workflow` is mocked because in
+    the test suite celery is run in eager mode. This prevent celery to switch
+    back from the api context to the app context, generating errors during
+    the workflow execution. The patched version of `continue_workflow` uses
+    the correct application context to run the workflow.
+    """
+    def continue_wf_patched_context(workflow_app):
+        workflow_object_class._custom_continue = workflow_object_class.continue_workflow
+
+        def custom_continue_workflow(self, *args, **kwargs):
+            with workflow_app.app_context():
+                self._custom_continue(*args, **kwargs)
+
+        return custom_continue_workflow
+
+    record = record_from_db
+    del record['arxiv_eprints']
+    rec_id = record['control_number']
+
+    eng_uuid = start('article', [record])
+    obj_id = WorkflowEngine.from_uuid(eng_uuid).objects[0].id
+    obj = workflow_object_class.get(obj_id)
+
+    assert obj.status == ObjectStatus.HALTED  # for matching approval
+    obj_id = WorkflowEngine.from_uuid(eng_uuid).objects[0].id
+    obj = workflow_object_class.get(obj_id)
+
+    assert obj.extra_data['already-in-holding-pen'] is False
+    assert obj.extra_data['holdingpen_matches'] == []
+    assert obj.extra_data['matches']['fuzzy'] == [rec_id]
+
+    WorkflowObject.continue_workflow = continue_wf_patched_context(workflow_app)
+    do_resolve_matching(workflow_app, obj.id, rec_id)
+
+    obj = workflow_object_class.get(obj_id)
+    assert obj.extra_data['matches']['approved'] == rec_id
+    assert obj.extra_data['is-update']
```

Comment:
Should this be approved/completed? It's not really clear to me what we are testing here though :/, we tend to mix integration with end-to-end with acceptance with ui...

Reply:
This is testing a new branch of the workflow: when the fuzzy matching branch is walked through, some things happen in the workflow (marks, halting).